oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/query/AbstractTopologyTest.java

```diff
@@ -0,0 +1,230 @@
+        assertEquals(33, call2.getTarget().intValue());
+
+        Call call3 = topology.getCalls().get(3);
+        assertEquals(101, call3.getSource().intValue());
+        assertEquals(111, call3.getTarget().intValue());
+        Call call4 = topology.getCalls().get(4);
+        assertEquals(102, call4.getSource().intValue());
+        assertEquals(122, call4.getTarget().intValue());
+        Call call5 = topology.getCalls().get(5);
+        assertEquals(103, call5.getSource().intValue());
+        assertEquals(133, call5.getTarget().intValue());
+    }
+
+    protected List<Call.CallDetail> mockClientCall() {
+
+        List<Call.CallDetail> result = new ArrayList<>(10);
+
+        Call.CallDetail call0 = new Call.CallDetail();
+        call0.setSource(0);
+        result.add(call0);
+
+        Call.CallDetail call1 = new Call.CallDetail();
+        call1.setSource(1);
+        call1.setComponentId(1);
+        call1.setTarget(11);
+        ServiceInventory inventorySource1 = mock(ServiceInventory.class);
+        when(inventorySource1.getSequence()).thenReturn(11);
+        when(inventorySource1.getIsAddress()).thenReturn(1);
+
+        ServiceInventory inventoryTarget1 = mock(ServiceInventory.class);
+        when(inventoryTarget1.getIsAddress()).thenReturn(1);
+        when(serviceInventoryCache.get(1)).thenReturn(inventorySource1);
+        when(serviceInventoryCache.get(11)).thenReturn(inventoryTarget1);
+
+        Call.CallDetail call2 = new Call.CallDetail();
+        call2.setSource(2);
+        call2.setComponentId(2);
+        call2.setTarget(22);
+
+        ServiceInventory inventorySource2 = mock(ServiceInventory.class);
+        when(inventorySource2.getSequence()).thenReturn(22);
+        when(inventorySource2.getIsAddress()).thenReturn(0);
+
+        ServiceInventory inventoryTarget2 = mock(ServiceInventory.class);
+        when(serviceInventoryCache.get(2)).thenReturn(inventorySource2);
+        when(serviceInventoryCache.get(22)).thenReturn(inventoryTarget2);
+
+        Call.CallDetail call3 = new Call.CallDetail();
+        call3.setSource(3);
+        call3.setComponentId(3);
+        call3.setTarget(33);
+
+        ServiceInventory inventorySource3 = mock(ServiceInventory.class);
+        when(inventorySource3.getSequence()).thenReturn(33);
+        when(inventorySource3.getIsAddress()).thenReturn(1);
+
+        ServiceInventory inventoryTarget3 = mock(ServiceInventory.class);
+        when(serviceInventoryCache.get(3)).thenReturn(inventorySource3);
+        when(serviceInventoryCache.get(33)).thenReturn(inventoryTarget3);
+
+        Call.CallDetail call4 = new Call.CallDetail();
+        call4.setSource(4);
+        call1.setComponentId(4);
+        call4.setTarget(44);
+
+        ServiceInventory inventorySource4 = mock(ServiceInventory.class);
+        ServiceInventory inventoryTarget4 = mock(ServiceInventory.class);
+        when(inventoryTarget4.getMappingServiceId()).thenReturn(1);
```

Comment:
Also mapping means client and server sides have agents, so the `client -> peer` could be translated to `client -> server`. So, I need to describe the topology clear. Otherwise, it is hard to say that, your test cases are right and scenarios coveraged.